src/ui/panels/clan.js

```diff
@@ -99,20 +103,36 @@ export class Clan {
     }
 
     // Add clan member to list
-    addClanMember(member) {
+    addClanMember(memberName) {
         // Add to saved data
-        characterData.addClanMember(member.name);
+        characterData.addClanMember(memberName);
 
         // Add visuals if the dashboard is up
         if (this.dashboard.scene.isActive()) {
             const startX = 560,
                 startY = 280,
                 yDiff = 18;
 
-            // Add text to the list
-            let index = this.clanMembers.length;
-            member.createText(false, startX, startY + index * yDiff);
-            this.clanMembers.push(member);
+            // Add new clan entry if new member else increment existing member by one
+            const index = this.clanMembers.findIndex((member) => {
+                if (member.name === memberName) {
+                    return true;
+                }
+                return false;
+            });
+
+            if (index != -1) {
+                this.clanMembers[index].numberOwned++;
```

Comment:
```js
const existingMember = this.clanMembers.find((member) => member.name === memberName);
if (existingMember) {
  existingMember.numberOwned++;
}
```

Reply:
Also since we're changing the save data for members to be to be a dictionary maybe we should do the same for the game objects too

Reply:
Decided not to change that to a dictionary as I don't currently see a need for it. If we ever need to retrieve them by name then we can change it

Reply:
Decided not to change to a dict for now. If it is more convenient later on then we can update